pyomo/contrib/pynumero/src/ma27Interface.cpp

```diff
@@ -0,0 +1,285 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+// This would normally be in a header file, but as we do not need one,
+// we will explicitly include it here.
+#if defined(_WIN32) || defined(_WIN64)
+#  if defined(BUILDING_PYNUMERO_ASL)
+#    define PYNUMERO_HSL_EXPORT __declspec(dllexport)
+#  else
+#    define PYNUMERO_HSL_EXPORT __declspec(dllimport)
+#  endif
+#else
+#  define PYNUMERO_HSL_EXPORT
+#endif
+
+// Forward declaration of MA27 fortran routines
+extern "C" {
+   void ma27id_(int* ICNTL, double* CNTL);
```

Comment:
If I'm reading the commit history correctly, you removed this `extern "C"` then re-added it. I added this after googling "how to declare external fortran functions in c++." Do you know the real reason this should or shouldn't be necessary?

Reply:
I didn't think originally that you had to wrap the Fortran functions in "extern C", but of course you do: the names used in those declarations have to match the name mangling used by the fortran compiler (so we need the simpler, C-style mangling so that things work out).